app/Jobs/ExternalServices/IssueSecondaryForumGroup.php

```diff
@@ -32,49 +34,53 @@ public function __construct(int $cid, int $forumGroup)
      */
     public function handle()
     {
-        $ipboard = new Ipboard();
+        Redis::throttle('issue-secondary-forum-group')->allow(15)->every(60)->then(function () {
```

Comment:
What is the 15 represent?

Reply:
15 instances of this task can be ran every 60 seconds. It throttles the number of times we can physically run this job. It is designed to stop you hammering an API, but in our case, it should stop us hammer our own forum database.

Reply:
Can you put that into a semi-descriptive variable to improve readability? Something like `$maximumTaskAllowance = 15`

Reply:
I'm not sure that helps?
Its a documented feature here: https://laravel.com/docs/7.x/queues#rate-limiting

And it currently reads "throttle, allow 15, every 60" which seems quite readable to me?

Reply:
6 vs half-dozen. I'd like to be able to read that at first glance without needing the docs. I'm easy either way, but I didn't know what it was till I read the docs myself.

Reply:
I think extracting a var out for this feels a bit odd. I'd rather stick to convention that is outlined in the docs

Reply:
Can I wade in? Why don't you extract this whole thing into a middleware?

`<?php

namespace App\Jobs\Middleware;

use Illuminate\Support\Facades\Redis;

class RateLimited
{
    /** @var null */
    private $key;

    /** @var int */
    private $allow;

    /** @var int */
    private $every;

    /** @var int */
    private $retryAfter;

    public function __construct(string $key = null, ?int $allow = 10, ?int $every = 60, ?int $retryAfter = 10)
    {
        $this->key        = $key ?? 'rate_limited_queue_job';
        $this->allow      = $allow ?? 10;
        $this->every      = $every ?? 60;
        $this->retryAfter = $retryAfter ?? 10;
    }

    public function handle($job, $next)
    {
        Redis::throttle($this->key)
             ->allow($this->allow)
             ->every($this->every)
             ->then(function () use ($job, $next) {
                 // Lock obtained...

                 $next($job);
             }, function () use ($job) {
                 // Could not obtain lock...

                 $job->release($this->retryAfter);
             });
    }
}`

Then you can name your constructor argument whatever you want, without needing to repeat your throttling on every job => https://laravel.com/docs/7.x/queues#job-middleware

Reply:
Good shout! Thanks 👍